django/db/models/base.py

```diff
@@ -1117,9 +1117,9 @@ def _save_table(
             and not force_insert
             and not force_update
             and self._state.adding
-            and (
-                (meta.pk.default and meta.pk.default is not NOT_PROVIDED)
-                or (meta.pk.db_default and meta.pk.db_default is not NOT_PROVIDED)
+            and all(
+                f.has_default() or f.db_default is not NOT_PROVIDED
```

Comment:
Makes sense. FWIW I added the equivalent of `f.has_default()` but for `db_default` in [this PR](https://github.com/django/django/pull/18909/commits/e197a2b222696f246bc797d77b40267c702e8ae6) so we should eventually be able to get rid of this _ugly_ double negation and do

```suggestion
                f.has_default() or f.has_db_default()
```

Reply:
Created https://github.com/django/django/pull/19021 to include `has_db_default` if we want to get it in as well.